### PR TITLE
Fix shared dependency bug

### PR DESCRIPTION
**Pull request by gingermusketeer**

### CODE REVIEW

File: test/unit/check_modules_test.js [hunk 1]
```diff
@@ -6,27 +6,42 @@ var expect = require('chai').expect
...
 
   describe('when they are valid', function () {
     it('does not throw an exception', function () {
-      var packageInfo = require('../fixtures/valid_dependencies/package.json')
-      checkModules(
-        packageInfo,
-        path.resolve(fixture_path, 'valid_dependencies', 'node_modules')
-      )
+      fixtureName = 'valid_dependencies'
+      run()
```
Comment (ashoda) [inline test/unit/check_modules_test.js]: Run wasn't immediately obvious as far as what it does. May consider renaming it to expose intent

File: index.js [hunk 2]
```diff
@@ -7,6 +7,10 @@ module.exports = {
   validate: function validate (packageJsonPath) {
     var packageInfo = require(packageJsonPath)
     checkEngine(packageInfo)
-    checkModules(packageInfo, path.dirname(packageJsonPath) + '/node_modules')
+    checkModules(
+      packageInfo,
+      path.join(path.dirname(packageJsonPath), 'node_modules'),
+      ['development', 'test'].indexOf(process.env.NODE_ENV) >= 0
```
Comment (ashoda) [inline index.js]: I would move this up and assign it to a var. And possibly even LN#12

**Comment by ashoda** [comment]
A couple of comments but LGTM otherwise.

**Comment by adamcohen** [comment]
This still doesn't work for me.  Even after rm -rf'ing my node_modules dir, I get: 

    UnsavedDependenciesError: Unsaved dependencies installed: chai, chai-subset, 
    eslint, istanbul, mocha, pre-commit, rewire, sinon, sinon-chai, standard, supertest. 
    
    Run `npm prune` to remove them or add them to your package.json
    at checkModules (/Users/adam/vmshare/consumption_service/node_modules/
    @westfieldlabs/environment_validator/lib/check_modules.js:142:11)

All of those modules exist in my `package.json`

**Pull request merged**